Twitch/Twitch.Base/Services/NewAPI/WebhooksService.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Twitch.Base.Models.NewAPI.Webhooks;
+
+namespace Twitch.Base.Services.NewAPI
+{
+    /// <summary>
+    /// The APIs for Webhook-based services.
+    /// </summary>
+    public class WebhooksService : NewTwitchAPIServiceBase
+    {
+
+        /// <summary>
+        /// Creates an instance of the WebhooksService
+        /// </summary>
+        /// <param name="connection"></param>
+        public WebhooksService(TwitchConnection connection) : base(connection) { }
+
+        /// <summary>
+        /// Gets the Webhook subscriptions of a user identified by a Bearer token, in order of expiration.  Requires App Token.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<IEnumerable<WebhookSubscriptionModel>> GetWebhookSubscriptions(int maxResults=100)
+        {
+            if (maxResults < 10)
+                maxResults = 10;
+            if (maxResults > 100)
+                maxResults = 100;
+            return await this.GetPagedDataResultAsync<WebhookSubscriptionModel>("webhooks/subscriptions", maxResults);
+        }
+
+        /// <summary>
+        /// Subscribe to a webhook topic
+        /// </summary>
+        /// <param name="callback">URL where notifications will be delivered.</param>
+        /// <param name="topic">URL for the topic to subscribe to or unsubscribe from. topic maps to a new Twitch API endpoint.</param>
+        /// <param name="lease_seconds">Number of seconds until the subscription expires. Default: 0. Maximum: 864000.</param>
+        /// <param name="secret">Secret used to sign notification payloads. The X-Hub-Signature header is generated by sha256(secret, notification_bytes).</param>
+        /// <returns></returns>
+        public async Task<bool> SubscribeWebhookTopic(string callback, string topic, int lease_seconds, string secret)
+        {
+            return await processWebhookTopicRequest(callback,"subscribe",topic, lease_seconds, secret);
```

Comment:
Add Validator.ValidateString() & Validator.Validate() calls here to check input values

Reply:
completed